### PR TITLE
[SPARK-14814][MLlib] API: Java compatibility, docs

### PR DESCRIPTION
**Pull request by hhbyyh**
## What changes were proposed in this pull request?

jira: https://issues.apache.org/jira/browse/SPARK-14814
fix a java compatibility function in mllib DecisionTreeModel. As synced in jira, other compatibility issues don't need fixes. 
## How was this patch tested?

existing ut

### CODE REVIEW

File: mllib/src/test/java/org/apache/spark/mllib/tree/JavaDecisionTreeSuite.java [hunk 1]
```diff
@@ -21,6 +21,8 @@
 import java.util.HashMap;
 import java.util.List;
 
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.mllib.linalg.Vector;
```
Comment (HyukjinKwon) [inline mllib/src/test/java/org/apache/spark/mllib/tree/JavaDecisionTreeSuite.java]: I guess we need to reorder imports (See https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide#SparkCodeStyleGuide-Imports)

File: mllib/src/test/java/org/apache/spark/mllib/tree/JavaDecisionTreeSuite.java [hunk 2]
```diff
@@ -95,6 +97,14 @@ public void runDTUsingStaticMethods() {
 
     DecisionTreeModel model = DecisionTree$.MODULE$.train(rdd.rdd(), strategy);
 
+    // java compatibility test
+    JavaRDD<Double> predictions = model.predict(rdd.map(new Function<LabeledPoint, Vector>() {
+      @Override
+      public Vector call(LabeledPoint v1) throws Exception {
```
Comment (srowen) [inline mllib/src/test/java/org/apache/spark/mllib/tree/JavaDecisionTreeSuite.java]: Nit: no "throws Exception". Can you omit the types above and write new Function<>?

**Comment by hhbyyh** [comment]
Thanks for the review @srowen @HyukjinKwon . Updated according to the comments.

**Comment by srowen** [comment]
Merged to master/2.0

**Pull request closed without merging**